### PR TITLE
Encrypt the customized approval link url

### PR DESCRIPTION
**Pull request by bzwei**
combine the random access key and username and encrypt them as part of the main url.

https://projects.engineering.redhat.com/browse/SSP-1290

Current approval url that is clickable in the email sent to each approver looks like: https://ci.cloud.redhat.com/api/approval/v1.0/stageaction/ec0150c649d9b5fbfb74439eeaaffc35?approver=Z21jY3VsbG9AcmVkaGF0LmNvbQ==

After the work the url will look like https://ci.cloud.redhat.com/api/approval/v1.0/stageaction/v2%3A%7BCUmBLmGaC%2BofRZENlA7lHf1ZcMf%2Flmma0ISE3rmfk7LudMWJ5vDvvEi6Qq8TCdDwvRVgM9g0v5Qf1KSRsxBVcw%3D%3D%7D

### CODE REVIEW

File: src/main/java/com/redhat/management/approval/EmailBody.java [hunk 1]
```diff
@@ -170,6 +176,26 @@ private static String getUrlContent(URL url) throws Exception {
         return content.toString();
     }
 
+    private String encrypt(String strToEncrypt) throws Exception {
+        String v2Key = System.getenv("ENCRYPTION_KEY");
```
Comment (mkanoor) [inline src/main/java/com/redhat/management/approval/EmailBody.java]: @bzwei do we need to add a task to add this into
https://console.insights-dev.openshift.com/console/project/approval-ci/browse/dc/approval-pam?tab=environment

File: src/main/java/com/redhat/management/approval/EmailBody.java [hunk 2]
```diff
@@ -170,6 +176,26 @@ private static String getUrlContent(URL url) throws Exception {
...
+        byte[] keyBytes = Base64.getDecoder().decode(v2Key);
+        SecretKeySpec secretKey = new SecretKeySpec(keyBytes, "AES");
+
+        byte[] iv = { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+        IvParameterSpec ivpec = new IvParameterSpec(iv);
+
+        Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
+        cipher.init(Cipher.ENCRYPT_MODE, secretKey, ivpec);
+
+        String output = Base64.getEncoder().encodeToString(cipher.doFinal(strToEncrypt.getBytes("UTF-8")));
```
Comment (mkanoor) [inline src/main/java/com/redhat/management/approval/EmailBody.java]: @bzwei Should you be using the URL Encoder which doesn't add new lines
https://docs.oracle.com/javase/8/docs/api/java/util/Base64.html

File: src/main/java/com/redhat/management/approval/EmailBody.java (hunk 1 above)
Comment (bzwei) [reply src/main/java/com/redhat/management/approval/EmailBody.java]: https://github.com/RedHatInsights/e2e-deploy/pull/1145

File: src/main/java/com/redhat/management/approval/EmailBody.java (hunk 2 above)
Comment (bzwei) [reply src/main/java/com/redhat/management/approval/EmailBody.java]: I followed the way how `ManageIQ::Password` encrypts: https://github.com/ManageIQ/manageiq-password/blob/master/lib/manageiq/password.rb#L26. It has to be decrypt precisely in our ruby code.

**Comment by bzwei** [comment]
Will take another approach.

**Pull request closed without merging**